crates/router/src/connector/noon/transformers.rs

```diff
@@ -410,14 +411,20 @@ impl<F, T>
                     reason: Some(error_message),
                     status_code: item.http_code,
                 }),
-                _ => Ok(types::PaymentsResponseData::TransactionResponse {
-                    resource_id: types::ResponseId::ConnectorTransactionId(order.id.to_string()),
-                    redirection_data,
-                    mandate_reference,
-                    connector_metadata: None,
-                    network_txn_id: None,
-                    connector_response_reference_id: None,
-                }),
+                _ => {
+                    let connector_response_reference_id =
+                        Some(order.reference.unwrap_or(order.id.to_string()));
```

Comment:
```suggestion
                    let connector_response_reference_id =
                         order.reference.or(Some(order.id.to_string()));
```

You are unwrapping and then if the value is `Some` we are wrapping it again, whereas we can use `or` to reduce that one redundant operation

Reply:
Done @SamraatBansal